core/src/main/java/co/cask/wrangler/steps/transformation/RowHash.java

```
@@ -0,0 +1,88 @@
+)
+public class RowHash extends AbstractStep {
+    private final String column;
+    private final String codec;
+
+    public RowHash(int lineno, String directive, String column, String codec) {
+        super(lineno, directive);
+        this.column = column;
+        this.codec = codec.toLowerCase();
+    }
+
+    /**
+     * Executes a wrangle step on single {@link Record} and return an array of wrangled {@link Record}.
+     *
+     * @param records  Input {@link Record} to be wrangled by this step.
+     * @param context {@link PipelineContext} passed to each step.
+     * @return Wrangled {@link Record}.
+     */
+    @Override
+    public List<Record> execute(List<Record> records, PipelineContext context) throws StepException {
+        for (Record record : records) {
+            switch (codec) {
+                case "md5": {
+                    String md5 = DigestUtils.md5Hex(record.toString()).toUpperCase();
+                    record.addOrSet(column, md5);
+                }
+                break;
+
+                case "sha1": {
+                    String sha1 = DigestUtils.sha1Hex(record.toString()).toUpperCase();
+                    record.addOrSet(column, sha1);
+                }
+                break;
+
+                case "sha256": {
+                    String sha256 = DigestUtils.sha256Hex(record.toString()).toUpperCase();
+                    record.addOrSet(column, sha256);
+                }
+                break;
+
+                case "sha384": {
+                    String sha384 = DigestUtils.sha384Hex(record.toString()).toUpperCase();
+                    record.addOrSet(column, sha384);
+                }
+                break;
+
+                case "sha512": {
+                    String sha512 = DigestUtils.sha512Hex(record.toString()).toUpperCase();
+                    record.addOrSet(column, sha512);
+                }
+                break;
+
+                default: {
```

should we rather throw an exception if the codec is unknown?